codecarbon/core/cpu.py

```diff
@@ -231,31 +211,142 @@ def get_cpu_details(self) -> Dict:
         return cpu_details
 
 
+
 class TDP:
     def __init__(self):
-        self.model, self.tdp = self._get_power_from_constant()
+        self.model, self.tdp = self._main()
+
+
+    @staticmethod
+    def _detect_cpu_model() -> str:
+        try:
+            cpu_info = cpuinfo.get_cpu_info()
+            cpu_model_detected = cpu_info.get("brand_raw", "")
+            return cpu_model_detected
+        except:
+            return None
+
+
+    @staticmethod
+    def _get_cpu_constant_power(match:str, cpu_power_df:pd.DataFrame) -> int:
+        """ Extract constant power from matched CPU """
+        return cpu_power_df[cpu_power_df['Name'] == match]['TDP'].values[0]
+
+
+    def _get_cpu_power_from_registry(self, cpu_model_raw:str) -> int:
+        cpu_power_df = DataSource().get_cpu_power_data()
+        cpu_matching = self._get_matching_cpu(cpu_model_raw, cpu_power_df)
+        if cpu_matching:
+            power = self._get_cpu_constant_power(cpu_matching, cpu_power_df)
+            return power
+        else:
+            return None
+
+
+    @staticmethod
+    def _get_cpus(cpu_df, cpu_idxs) -> list:
+        return [cpu_df['Name'][idx] for idx in cpu_idxs]
+
+
+    @staticmethod
+    def _get_direct_matches(moodel:str, cpu_df:pd.DataFrame) -> list:
+        model_l = moodel.lower()
+        return [fuzz.ratio(model_l, cpu.lower()) for cpu in cpu_df['Name']]
+
+
+    @staticmethod
+    def _get_token_set_matches(model:str, cpu_df:pd.DataFrame) -> list:
+        return [fuzz.token_set_ratio(model, cpu) for cpu in cpu_df['Name']]
+
+
+    @staticmethod
+    def _get_single_direct_match(ratios:list, max_ratio:int, cpu_df:pd.DataFrame) -> str:
+        idx = ratios.index(max_ratio)
+        cpu_matched = cpu_df['Name'].iloc[idx]
+        return cpu_matched
 
-    def _get_power_from_constant(self) -> int:
+
+    def _get_matching_cpu(self, model_raw:str, cpu_df:pd.DataFrame,
+                          threshold_direct=100, threshold_token_set=100,
```

Comment:
i guess the thresholds could be private vars, there is no need for them to be set outside of this function (this could lead to inconsistencies if a user choses to update those values and the match differs from its usual CPU).
Wdyt ?

Reply:
Yes I agree! Thanks

Reply:
I'm now thinking on the contrary that we could set these variables as arguments in the main function. This could help the user force a match when it has not matched automatically. For example, one could set _greedy_ to True, or lower the _thresholds_ in order to force the match of a cpu. The user could then check by itself whether the cpu has a close tdp to its actual cpu.

Reply:
I don't think it should be the user's responsibility to manage the cpu match, but if you do then your description seems a good way to do it.

Reply:
Ok I've left it as private vars!